lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -81,6 +82,19 @@ def scrub(transcript)
 
       private
 
+      def validation(options = {})
+        post = {}
+        add_3ds_validation_params(post, options)
+        commit(:validation, post, options)
+      end
+
+      def complete_transaction(action, amount, payment_method, options = {})
+        MultiResponse.run do |r|
+          r.process { validation(options) }
+          r.process { add_transaction_params_commit(action, amount, {}, payment_method, options.merge!(validation: r.params)) }
```

Comment:
💬 Comment:

Maybe the name `add_transaction_params_commit` is not entirely accurate because this method does way more than just that